### PR TITLE
#63 - Allow to start a feature/hotfix from a non default branch

### PR DESCRIPTION
**Pull request by alextsc**
Hey,

this is a implementation of the feature request in https://github.com/OpherV/gitflow4idea/issues/63.

The new feature and new hotfix dialogs have been replaced with a new implementation that allows to pick a branch on which the new branch is based on.

newfeature

newfeature2

Sorry about the diff mess down below, I just saw that the autoformatter took the liberty to make a few adjustments when it comes to import ordering, whitespace and the like in some of the older files (Gitflow interface, GitflowImpl, ...). I dont think its a major issue but if you want me to revert those files back to the old style just say the word. 

Alex

### CODE REVIEW

File: src/gitflow/GitflowImpl.java [hunk 1]
```diff
@@ -378,13 +390,28 @@ public GitCommandResult publishHotfix(@NotNull GitRepository repository,
         return run(h);
     }
 
-    private void setUrl (GitLineHandler h, GitRepository repository){
-        ArrayList<GitRemote> remotes = new ArrayList(repository.getRemotes());
+    @Override
+    public List<String> getBranchList(GitRepository repository) {
```
Comment (OpherV) [inline src/gitflow/GitflowImpl.java]: How about using getLocalBranchNames in GitflowBranchUtil where needed instead of adding this here?

Comment (alextsc) [inline src/gitflow/GitflowImpl.java]: Yeah you're right, I've seen that method too late. Will adjust it later. :)

File: src/gitflow/ui/AbstractBranchStartDialog.java [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+        branchList.remove(defaultBranch);
+
+        ComboEntry[] entries = new ComboEntry[branchList.size() + 1];
+        entries[0] = new ComboEntry(defaultBranch, defaultBranch + " (default)");
+        for (int i = 1; i <= branchList.size(); i++) {
+            String branchName = branchList.get(i - 1);
+            entries[i] = new ComboEntry(branchName, branchName);
+        }
+
+        return new DefaultComboBoxModel<>(entries);
```
Comment (OpherV) [inline src/gitflow/ui/AbstractBranchStartDialog.java]: Diamond type

File: src/gitflow/GitflowImpl.java [hunk 3]
```diff
@@ -378,13 +390,14 @@ public GitCommandResult publishHotfix(@NotNull GitRepository repository,
         return run(h);
     }
 
-    private void setUrl (GitLineHandler h, GitRepository repository){
-        ArrayList<GitRemote> remotes = new ArrayList(repository.getRemotes());
+    private void setUrl(GitLineHandler h, GitRepository repository) {
+        ArrayList<GitRemote> remotes = new ArrayList<>(repository.getRemotes());
```
Comment (OpherV) [inline src/gitflow/GitflowImpl.java]: Diamond type

**Comment by OpherV** [comment]
@alextsc After merging I've noticed you used diamond types which are Java 7 features (I've marked where). From experience, compiling the plugin using a JDK higher than 1.6 causes problems with users who only have the 1.6 SDK. Can you make sure to use only 1.6 features and submit a fix?

**Comment by OpherV** [comment]
Nevermind - I did the fix myself :D

**Comment by alextsc** [comment]
Oh wow, you're right. Damn habbit. Sorry about that too. =/

**Pull request merged**